server/pipeline_trigger.go

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package server
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+)
+
+func getPipelineFormData(pipelineTrigger *PipelineTrigger, args []string) url.Values {
+	data := url.Values{}
+	data.Add("token", pipelineTrigger.Token)
+	data.Add("ref", pipelineTrigger.Reference)
+	for key, value := range pipelineTrigger.Variables {
+		var v = value
+		if search := strings.TrimPrefix(value, "%%"); search != value {
+			for _, r := range args {
+				if replacement := strings.TrimPrefix(r, search+"="); replacement != r {
+					v = replacement
+					break
+				}
+			}
+		}
+		data.Add(fmt.Sprintf("variables[%s]", key), v)
+	}
+	return data
+}
+
+func post(url string, formData url.Values) (map[string]interface{}, error) {
+	response, err := http.PostForm(url, formData)
+	if err != nil {
+		return nil, err
+	}
+	defer response.Body.Close()
+	responseData, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return nil, err
+	}
+	if response.StatusCode < 200 || response.StatusCode >= 300 {
+		return nil, fmt.Errorf("invalid request = %s,%s", response.Status, string(responseData))
+	}
+
+	var result map[string]interface{}
+	json.Unmarshal(responseData, &result)
```

Comment:
Should we handle the possibility that the `responseData` isn't valid JSON? Otherwise, `result` will be `nil` and we could end up segfaulting elsewhere silently.

Reply:
Invalid json case is handled
https://github.com/mattermost/matterbuild/pull/92/files#diff-3a2928403027668f297d9bce542b27fe04a47b86650a36900a1e408018da70e5R50-R52